### PR TITLE
Potential Privilege Escalation BugFix 

### PR DESCRIPTION
**Pull request by krishna619**
What kind of change does this PR introduce?

BugFix

Issue Number:

Fixes https://github.com/PalisadoesFoundation/talawa-api/issues/1792

Did you add tests for your changes?

Yes.

Summary

This vulnerability could allow attackers to gain unauthorized access to user accounts, potentially leading to privilege escalation and compromise of sensitive data.
So, the token should be verified based on the original email address associated with the request.

Does this PR introduce a breaking change?
No

Other information

Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa-api/blob/master/CONTRIBUTING.md)?

### CODE REVIEW

File: README.md
Comment (SiddheshKukade) [inline README.md:1]: No need to make changes in this file

**Review by SiddheshKukade — CHANGES_REQUESTED** [review]
Fix the changes requested. Otherwise code is looking good to merge

File: README.md
Comment (krishna619) [reply README.md:1]: @SiddheshKukade 
on which file exactly do I have to make the change?

**Comment by Cioppolo14** [comment]
@krishna619 Please fix the failing tests.

**Comment by palisadoes** [comment]
We have a policy of unassigning contributors who close PRs without getting validation from our reviewer team. This is because:

1. We start looking for people to review PRs when you submit them. 
2. We often contact them and link to the PR. If the PR is closed the whole effort is wasted.
3. The historical thread of reviewer comments is broken when the work is spread across multiple PRs. The quality of our code is affected negatively.

Please be considerate of our volunteers' limited time and our desire to improve our code base.

This policy is stated as a pinned post in all our Talawa repositories. Our YouTube videos explain why this practice is not acceptable to our Community.

**Pull request closed without merging**